src/example-relay/src/graphql/locations/types/input/AddLocationInput.js

```diff
@@ -0,0 +1,29 @@
+// @flow strict
+
+import { GraphQLInputObjectType, GraphQLString, GraphQLEnumType, GraphQLNonNull } from 'graphql';
+
+const LocationType = new GraphQLEnumType({
+  name: 'LocationType',
+  values: {
+    AIRPORT: { value: 'airport' },
+    CITY: { value: 'city' },
+    COUNTRY: { value: 'country' },
+  },
+});
+
+export default new GraphQLInputObjectType({
+  name: 'AddLocationInput',
+  fields: {
+    locationId: {
+      type: GraphQLNonNull(GraphQLString),
+      description: '3-letter IATA code of airport or internal city code.',
+    },
+    name: {
```

Comment:
Why not make `name` and `type` required (non null)? IMHO it makes sense to require all those fields to create new location. And you don't have to worry about input validation in the mutation resolver.

Reply:
Yeah, I wanted to throw in here some examples of union return type, and some error handling. I was thinking about adding this later. But I guess I can make these non null, and still have some cases for showing of error handling.